openmdao/core/group.py

```diff
@@ -1992,101 +2004,27 @@ def _setup_connections(self):
-                                  f"indices expect {source_dimensions}."
-                            if self._raise_connection_errors:
-                                raise ValueError(msg)
-                            else:
-                                issue_warning(msg, category=SetupWarning)
-                                fail = True
-                                continue
-                    else:
-                        source_dimensions = 1
-
-                    # check all indices are in range of the source dimensions
-                    if flat or src_indices.ndim == 1:
-                        if allprocs_abs2meta_in[abs_in]['distributed']:
-                            out_size = np.sum(sizes_out[:, out_idxs[abs_out]])
+                    if src_indices.src_ndim > len(out_shape):
+                        msg = f"{self.msginfo}: The source indices " + \
+                              f"{meta_in['src_indices']} do not specify a " + \
+                              f"valid shape for the connection '{abs_out}' to '{abs_in}'. " + \
+                              f"The source has {len(out_shape)} dimensions but the " + \
+                              f"indices expect {src_indices.src_ndim}."
+                        if self._raise_connection_errors:
+                            raise ValueError(msg)
                         else:
-                            out_size = shape_to_len(out_shape)
-                        if src_indices.size > 0:
-                            mx = np.max(src_indices)
-                            mn = np.min(src_indices)
-                            if mx >= out_size:
-                                bad_idx = mx
-                            elif mn < -out_size:
-                                bad_idx = mn
-                            else:
-                                bad_idx = None
-                            if bad_idx is not None:
-                                msg = f"{self.msginfo}: The source indices do not specify " + \
-                                      f"a valid index for the connection '{abs_out}' to " + \
-                                      f"'{abs_in}'. Index '{bad_idx}' is out of range for " + \
-                                      f"source dimension of size {out_size}."
-                                if self._raise_connection_errors:
-                                    raise ValueError(msg)
-                                else:
-                                    issue_warning(msg, category=SetupWarning)
-                                    fail = True
-                        if src_indices.ndim > 1:
-                            meta_in['src_indices'] = src_indices.ravel()
+                            issue_warning(msg, category=SetupWarning)
+                            fail = True
+                            continue
+
+                    if src_indices.size != shape_to_len(in_shape):
```

Comment:
Is this the same check we're already doing on line 1990?

Reply:
yup.. removed